tests/4.5/target_teams_distribute/test_target_teams_distribute_dist_schedule.c

```diff
@@ -0,0 +1,98 @@
+//===------ test_target_teams_distribute_dist_schedule.c ------------------===//
+// 
+// OpenMP API Version 4.5 Nov 2015
+// 
+// This test checks that the dist_schedule clause (which must have kind 
+// static) correctly causes CHUNK_SIZE iterations to be split among the
+// number of teams the test is run with, in a round-robin faction in order
+// of the team number, when a chunk size is given. The test also confirms
+// that when no chunk size is given, that each team receives no more than
+// one chunk.
+// 
+//===----------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+#define CHUNK_SIZE 64
+
+int test_dist_schedule() {
+  int errors = 0;
+  int num_teams;
+  int a[N];
+  int b[N];
+
+  for (int i = 0; i < N; ++i) {
+    a[i] = -1;
+    b[i] = -1;
+  }
+
+#pragma omp target teams distribute map(from: num_teams) map(tofrom: a[0:N]) dist_schedule(static, CHUNK_SIZE)
+  for (int i = 0; i < N; ++i) {
+    if (omp_get_team_num() == 0) {
+      num_teams = omp_get_num_teams();
+    }
+    a[i] = omp_get_team_num();
+  }
+
+  if (num_teams == 1) {
+    OMPVV_WARNING("Cannot test dist_schedule(static, chunk_size) because num_teams was 1.");
+  } else if (num_teams < 1) {
+    OMPVV_ERROR("omp_get_num_teams() returned a value less than 1.");
+    return 1;
+  }
+
+  int counter = -1;
+  for (int i = 0; i < N; ++i) {
+    if (i % CHUNK_SIZE == 0) {
+      counter = (counter + 1) % num_teams;
+    }
+    if (a[i] != counter) {
+      OMPVV_ERROR("Loop iterations were not properly scheduled with specified chunk_size of %d.", CHUNK_SIZE);
+      errors++;
+      break;
+    }
+  }
+
+  num_teams = -1;
+
+#pragma omp target teams distribute map(from: num_teams) map(tofrom: b[0:N]) dist_schedule(static)
+  for (int i = 0; i < N; ++i) {
+    if (omp_get_team_num() == 0) {
+      num_teams = omp_get_num_teams();
+    }
+    b[i] = omp_get_team_num();
+  }
+
+  if (num_teams == 1) {
+    OMPVV_WARNING("Cannot test dist_schedule(static) because num_teams was 1.");
+  } else if (num_teams < 1) {
+    OMPVV_ERROR("omp_get_num_teams() returned a value less than 1.");
+    return 1;
+  }
+
+  for (int i = 1; i < N; ++i) {
```

Comment:
use omp_get_schedule to find out implementation defined and then compare.

Reply:
It turns out that the OMP_SCHEDULE environment variable only pertains to loops with schedule type `runtime`. Since we can only use `static` on a `distribute` clause, this method will not work.